e2e/api/api_test.go

```diff
@@ -186,3 +188,144 @@ func (s *APITestSuite) TestLogoutRemovesCells() {
+	require.NotNil(key.ExtendedKey, "CKD#2 has not been generated for new account")
+
+	// try creating sub-account, w/o selecting main account i.e. w/o login to main account
+	createSubAccountResponse := s.api.CreateChildAccount("", TestConfig.Account1.Password)
+	require.EqualValues(account.ErrNoAccountSelected, createSubAccountResponse.ErrorValue, "expected error is not returned (tried to create sub-account w/o login): %v", createSubAccountResponse.Error)
+
+	err = s.api.SelectAccount(address, TestConfig.Account1.Password)
+	require.NoError(err, "Test failed: could not select account: %v", err)
+
+	// try to create sub-account with wrong password
+	createSubAccountResponse = s.api.CreateChildAccount("", "wrong password")
+	require.EqualError(createSubAccountResponse.ErrorValue, "cannot retrieve a valid key for a given account: could not decrypt key with given passphrase", "expected error is not returned (tried to create sub-account with wrong password): %v", createSubAccountResponse.Error)
+
+	// create sub-account (from implicit parent)
+	createSubAccountResponse1 := s.api.CreateChildAccount("", TestConfig.Account1.Password)
+	require.NoError(createSubAccountResponse1.ErrorValue, "Test failed: could not select account: %v", createSubAccountResponse1.ErrorValue)
+
+	// make sure that sub-account index automatically progresses
+	createSubAccountResponse2 := s.api.CreateChildAccount("", TestConfig.Account1.Password)
+	require.NoError(createSubAccountResponse2.ErrorValue, "cannot create sub-account: %v", createSubAccountResponse2.ErrorValue)
+	require.NotEqual(createSubAccountResponse1.Address, createSubAccountResponse2.Address, "sub-account index auto-increament failed")
+	require.NotEqual(createSubAccountResponse1.PubKey, createSubAccountResponse2.PubKey, "sub-account index auto-increament failed")
+
+	// create sub-account (from explicit parent)
+	createSubAccountResponse3 := s.api.CreateChildAccount(createSubAccountResponse2.Address, TestConfig.Account1.Password)
+	require.NoError(createSubAccountResponse3.ErrorValue, "cannot create sub-account: %v", createSubAccountResponse3.ErrorValue)
+
+	subAccount1, subAccount2, subAccount3 := createSubAccountResponse1.Address, createSubAccountResponse2.Address, createSubAccountResponse3.Address
+	subPubKey1, subPubKey2, subPubKey3 := createSubAccountResponse1.PubKey, createSubAccountResponse2.PubKey, createSubAccountResponse3.PubKey
+
+	require.NotEqual(subAccount1, subAccount3, "sub-account index auto-increament failed: subAccount1 == subAccount3")
+	require.NotEqual(subPubKey1, subPubKey3, "sub-account index auto-increament failed: subPubKey1 == subPubKey3")
+	require.NotEqual(subAccount2, subAccount3, "sub-account index auto-increament failed: subAccount2 == subAccount3")
+	require.NotEqual(subPubKey2, subPubKey3, "sub-account index auto-increament failed: subPubKey2 == subPubKey3")
+
+	return true
+}
+
+func (s *APITestSuite) TestRecoverAccount() bool {
```

Comment:
Please compare this with e2e/accounts/accounts_test.go as a lot of test cases repeats here.